internal/configprovider/config_source_provider.go

```diff
@@ -109,6 +118,12 @@ func (c *configSourceConfigMapProvider) Get(context.Context) (*config.Map, error
 		return nil, err
 	}
 
+	// If we've already initiated and started a server, shut it down and create a new one for
+	// updated config sources.
+	if c.configServer != nil {
+		c.configServer.shutdown()
```

Comment:
Rather than cycling like this can we slightly alter the config server's handlers to render the initial and effective by reference at call time instead of marshaling beforehand?

Reply:
Sorry, but I'm not sure I understand your request here. I'm not following your thought as far as ordering of events. Could you expand on your thought `to render the initial and effective by reference at call time`?

Reply:
The initial and effective map references would be updated for each `Retrieve` call and the mux handlers would be updated to not use closures but instead marshal initial and redacted effective dynamically. I think the added overhead per request would be negligible.

```go
if c.configServer == nil {
    c.configServer = newConfigServer(c.logger)
    if err = c.configServer.start(); err != nil {
        return nil, err
    }
}

// these may need locks, as well as the updated mux handlers (not shown)*
c.configServer.setInitial(c.wrappedRetrieved.Map.ToStringMap())
c.configServer.setEffective(effectiveMap.ToStringMap())
```

Reply:
Updated to use locking to serve initial and effective configs by reference.